src/scriptsys/script.h

```diff
@@ -3,10 +3,35 @@
 
 #define SCRIPT_DIR "script/"
 
+#define LUA_FN(n) { .name = #n, .func = func_##n }
+
+#define LUA_LIB(L, ...) luaL_Reg lib[] = \
+{ \
+	__VA_ARGS__, \
+	{ 0 } \
+}; \
+luaL_newlib(L, lib);
+
+#define LUA_LIB_EMPTY(L) luaL_Reg lib[] = { { 0 } }; \
```

Comment:
Could these others be functions rather than macros?

Reply:
Haha, just after I closed out and was going to bed I was like "... wait" and realized that I was overdoing it with the macros.